src/UdpHubListener.cpp

```diff
@@ -123,145 +124,151 @@ UdpHubListener::~UdpHubListener()
 // the client is already on the thread pool, it means that a new connection is
 // requested (the old was desconnected). So we have to remove that thread from
 // the pool and then connect again.
-void UdpHubListener::run()
+void UdpHubListener::start()
 {
     mStopped = false;
 
-    QHostAddress PeerAddress; // Object to store peer address
-    int peer_udp_port; // Peer listening port
-    int server_udp_port; // Server assigned udp port
-
-    // Create and bind the TCP server
+    // Bind the TCP server
     // ------------------------------
-    QTcpServer TcpServer;
-    if ( !TcpServer.listen(QHostAddress::Any, mServerPort) ) {
-        QString error_message = QString("TCP Socket Server on Port %1 ERROR: %2").arg(mServerPort).arg(TcpServer.errorString());
+    QObject::connect(&mTcpServer, &QTcpServer::newConnection, this, &UdpHubListener::receivedNewConnection);
+    if ( !mTcpServer.listen(QHostAddress::Any, mServerPort) ) {
+        QString error_message = QString("TCP Socket Server on Port %1 ERROR: %2").arg(mServerPort).arg(mTcpServer.errorString());
         std::cerr << error_message.toStdString() << endl;
         emit signalError(error_message);
         return;
     }
+    
+    cout << "JackTrip HUB SERVER: Waiting for client connections..." << endl;
+    cout << "JackTrip HUB SERVER: Hub auto audio patch setting = " << mHubPatch 
+         << " (" << mHubPatchDescriptions.at(mHubPatch).toStdString() << ")" << endl;
+    cout << "=======================================================" << endl;
+    
+    // Start our monitoring timer
+    mStopCheckTimer.setInterval(200);
+    connect(&mStopCheckTimer, &QTimer::timeout, this, &UdpHubListener::stopCheck);
+    mStopCheckTimer.start();
+}
+    
+void UdpHubListener::receivedNewConnection()
+{
+    QTcpSocket *clientSocket = mTcpServer.nextPendingConnection();
+    connect(clientSocket, &QAbstractSocket::readyRead, this, &UdpHubListener::receivedClientInfo);
+    cout << "JackTrip HUB SERVER: Client Connection Received!" << endl;
+}
 
-    const int tcpTimeout = 5*1000;
-
-    cout << "JackTrip HUB SERVER: TCP Server Listening in Port = " << TcpServer.serverPort() << endl;
-    while (!mStopped && !sSigInt)
+void UdpHubListener::receivedClientInfo()
+{
+    QTcpSocket* clientConnection = static_cast<QTcpSocket*>(QObject::sender());
```

Comment:
This is sort of a hack from the viewpoint of Qt signal-slot mechanism. I think it's totally Ok here, but it could be avoided with connecting lambda expression instead of slot in `receivedNewConnection`.

Reply:
I don't really have a strong opinion either way, but a lambda might be the better option, and it's a fairly trivial fix. I could upload a change now or I could wait until I merge more of my developments over.

Reply:
I don't have a strong opinion either. It's totally up to you. I just wanted to mention this option hoping it might be helpful.